theano/compile/ops.py

```diff
@@ -675,6 +675,7 @@ def __str__(self):
     def make_node(self, x):
         if self.axis.keys() and (x.ndim <= max(self.axis.keys())):
             raise ValueError('Trying to rebroadcast non-existent dimension')
+        self.axis = OrderedDict([(k % x.ndim, v) if k < 0 else (k, v) for k, v in self.axis.items()])
```

Comment:
can you rename k, v to axis and broad as in the \__init__ method? This is more clear.

We do axis + x.ndim when we are in negative number. For consistency, can you do that here too instead of modulo?

This would also need tests. Normally, they should be in theano/compile/tests/test_ops.py, but for history and a few other reason, the current rebroadcast tests are in theano/tensor/tests/test_basic.py in the class test_broadcast. Can you add them there?

thanks for the PR.